tests/mocks.py

```diff
@@ -1037,6 +1105,67 @@ def to_response(i):
 
         return to_response(diffs[0])
 
+    @conduit_method("transaction.search")
+    def transaction_search(
+        self,
+        *,
+        objectIdentifier=None,
+        constraints={},
+        order=None,
+        before=None,
+        after=None,
+        limit=100,
+    ):
+        def to_response(i):
+            return deepcopy(
+                {
+                    "id": i["id"],
+                    "phid": i["phid"],
+                    "type": i["type"],
+                    "authorPHID": i["authorPHID"],
+                    "objectPHID": i["objectPHID"],
+                    "dateCreated": i["dateCreated"],
+                    "dateModified": i["dateModified"],
+                    "groupID": i["groupID"],
+                    "comments": i["comments"],
+                    "fields": i["fields"],
+                }
+            )
+
+        items = list(self._transactions)
+
+        if objectIdentifier:
```

Comment:
`objectIdentifier` is required by transaction search:
```json
{"result":null,"error_code":"ERR-CONDUIT-CORE","error_info":"When calling \"transaction.search\", you must provide an object to retrieve transactions for."}
```

See other examples in this file where we raise a `PhabricatorAPIException`